libraries/chain/include/scorum/chain/evaluators/proposal_evaluators.hpp

```diff
@@ -116,21 +116,30 @@ struct development_committee_transfer_evaluator
     void do_apply(const operation_type& o);
 };
 
-#define SCORUM_MAKE_TOP_BUDGET_AMOUNT_EVALUATOR_CLS_NAME(TYPE)                                                         \
-    BOOST_PP_SEQ_CAT((development_committee_change_top_)(TYPE)(_budgets_amount_evaluator))
-#define SCORUM_DEVELOPMENT_COMMITTEE_CHANGE_TOP_BUDGET_AMOUNT_EVALUATOR_DECLARE(TYPE)                                  \
-    struct SCORUM_MAKE_TOP_BUDGET_AMOUNT_EVALUATOR_CLS_NAME(TYPE)                                                      \
-        : public proposal_operation_evaluator<SCORUM_MAKE_TOP_BUDGET_AMOUNT_EVALUATOR_CLS_NAME(TYPE)>                  \
-    {                                                                                                                  \
-        typedef SCORUM_MAKE_TOP_BUDGET_AMOUNT_OPERATION_CLS_NAME(TYPE) operation_type;                                 \
-                                                                                                                       \
-        SCORUM_MAKE_TOP_BUDGET_AMOUNT_EVALUATOR_CLS_NAME(TYPE)(data_service_factory_i & r);                            \
-                                                                                                                       \
-        void do_apply(const operation_type& o);                                                                        \
-    };
-
-SCORUM_DEVELOPMENT_COMMITTEE_CHANGE_TOP_BUDGET_AMOUNT_EVALUATOR_DECLARE(post)
-SCORUM_DEVELOPMENT_COMMITTEE_CHANGE_TOP_BUDGET_AMOUNT_EVALUATOR_DECLARE(banner)
+template <typename OperationType, budget_type type>
```

Comment:
We are duplicating ourselves by providing the budget type information in both template parameters. I would like to make this `development_committee_change_top_budgets_amount_evaluator` template depending only on `budget_type` enum. In this case we need to make `development_committee_change_top_<...>_budgets_amount_operation` also template which depends on `budget_type` enum.